prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -161,6 +165,36 @@ def canonicalize_dataset_using_columns_for_split(
             {"input_col": input_col, "output_col": output_col}
         )
 
+    @staticmethod
+    def automatic_column_selection(
+        instruction: str, dataset_name: str, dataset_columns: str, example_rows: dict
+    ):
```

Comment:
Add type annotation.
```suggestion
    ) -> tuple[list[str], str]:
```